### PR TITLE
MODINVSTOR-674: Implement inventory view instances API

### PR DESCRIPTION
**Pull request by bohdan-suprun**
https://issues.folio.org/browse/MODINVSTOR-674 - Implement bulk instance endpoint

## Changes:
* Implement `inventory-view` API;
* Register new `instance_holdings_item_view` view.

### CODE REVIEW

**Review by marcjohnson-kint — COMMENTED** [review]
How is this different to the hierarchical view API?

File: descriptors/ModuleDescriptor-template.json [hunk 1]
```diff
@@ -1116,6 +1116,17 @@
...
+    {
+      "id": "inventory-view",
+      "version": "1.0",
+      "handlers": [
+        {
+          "methods": ["GET"],
+          "pathPattern": "/inventory-view/instances",
+          "permissionsRequired": ["inventory-storage.inventory-view.instances.collection.get"]
+        }
+      ]
```
Comment (marcjohnson-kint) [inline descriptors/ModuleDescriptor-template.json:1129]: How is this different to the existing inventory hierarchy API?

**Comment by hjiebsco** [comment]
> How is this different to the hierarchical view API?

@marc-johnson, looks like this one returns all fields from `instance` rather than just four fields (`instanceId, source, modeOfIssuance, natureOfContent`) defined by the [hierarchical view](https://github.com/folio-org/mod-inventory-storage/blob/master/src/main/resources/templates/db_scripts/inventory-hierarchy/createRecordsViewFunction.sql#L169-L179).

**Review by hjiebsco — APPROVED** [review]
LGTM. It would be nice if `InventoryViewTest` demonstrates returning more than one `instance` records. That is more likely to be case used by mod-search.

**Review by pfilippovEPAM — APPROVED** [review]
Looks good to me.

**Comment by bohdan-suprun** [comment]
> LGTM. It would be nice if `InventoryViewTest` demonstrates returning more than one `instance` records. That is more likely to be case used by mod-search.

Hi @hjiebsco, I have created https://issues.folio.org/browse/MODINVSTOR-684 for this, so that we can close this story till end of the sprint.

**Comment by marcjohnson-kint** [comment]
> > How is this different to the hierarchical view API?
> 
> @marc-johnson, looks like this one returns all fields from `instance` rather than just four fields (`instanceId, source, modeOfIssuance, natureOfContent`) defined by the [hierarchical view](https://github.com/folio-org/mod-inventory-storage/blob/master/src/main/resources/templates/db_scripts/inventory-hierarchy/createRecordsViewFunction.sql#L169-L179).

Thanks @hjiebsco 

I really should've been more explicit with my question.

Why was a new API introduced instead of extending an existing one that seems to serve the same purpose?

**Comment by bohdan-suprun** [comment]
> > > How is this different to the hierarchical view API?
> > 
> > 
> > @marc-johnson, looks like this one returns all fields from `instance` rather than just four fields (`instanceId, source, modeOfIssuance, natureOfContent`) defined by the [hierarchical view](https://github.com/folio-org/mod-inventory-storage/blob/master/src/main/resources/templates/db_scripts/inventory-hierarchy/createRecordsViewFunction.sql#L169-L179).
> 
> Thanks @hjiebsco
> 
> I really should've been more explicit with my question.
> 
> Why was a new API introduced instead of extending an existing one that seems to serve the same purpose?

@marcjohnson-kint, the existing view SQL is too complex to update it especially before module releases. Also, I'm not sure about it's performance, because it retrieves some additional records that are not needed e.g. for search. 

We might want to merge this two APIs at some future. But I'd recommend to proceed with this new API for now, if that's fine.

**Comment by marcjohnson-kint** [comment]
> > > > How is this different to the hierarchical view API?
> > > 
> > > 
> > > @marc-johnson, looks like this one returns all fields from `instance` rather than just four fields (`instanceId, source, modeOfIssuance, natureOfContent`) defined by the [hierarchical view](https://github.com/folio-org/mod-inventory-storage/blob/master/src/main/resources/templates/db_scripts/inventory-hierarchy/createRecordsViewFunction.sql#L169-L179).
> > 
> > 
> > Thanks @hjiebsco
> > I really should've been more explicit with my question.
> > Why was a new API introduced instead of extending an existing one that seems to serve the same purpose?
> 
> the existing view SQL is too complex to update it especially before module releases. 

I understand the need for an urgent resolution given the timing.

That is a little concerning with respect to the long term maintenance of the module. Out of curiosity, why do you consider the existing view too complex to change?

> We might want to merge this two APIs at some future. But I'd recommend to proceed with this new API for now, if that's fine.

That isn't my decision. FOLIO devolved technical design decisions to individual teams, which makes this Falcon's decision.

**Comment by bohdan-suprun** [comment]
@marcjohnson-kint, I'm going to merge the changes, because they are blockers for release. Feel free to leave comments here on in slack channel `#﻿elastic-search-backend-integration`

**Comment by bohdan-suprun** [comment]
> Out of curiosity, why do you consider the existing view too complex to change?

The SQL file has already 400+ lines, which most likely is not covered enough with tests.

**Pull request merged**